[WIP] Updated app for security

rorybee opened the pull request:
- Updated gradle plugin
- Removed deprecated Jack compiler options
- Added linter checks
- Updated compileSdkVersion to 30
- Updated targetSdkVersion to 30
- Fixed errors
- Fixed deprecated warnings
- Refactored - Migrated to AndroidX
- Added /.idea/caches to gitignore
- Updated versionCode
- Updated versionName

rorybee committed:
[WIP] Updated app for security

- Updated gradle plugin
- Removed deprecated Jack compiler options
- Added linter checks
- Updated compileSdkVersion to 30
- Updated targetSdkVersion to 30
- Fixed errors
- Fixed deprecated warnings
- Refactored - Migrated to AndroidX
- Added /.idea/caches to gitignore
- Updated versionCode
- Updated versionName

rorybee committed:
[WIP] Updated app for security

- Updated buildToolsVersion
- Added foreground service permission
- Fixed wakelock tag errors
- Fixed Handler deprecation warnings
- Fixed PreferenceManager deprecation warnings
- Fixed Notification.Builder(context) deprecation warning
- Fixed CheckBoxPreference deprecation warning
- Fixed PreferenceFragment deprecation warning
- Fixed return type warnings
- Fixed type conversion warnings
- Added "-Werror" linter check to fix warnings

grmpl commented:
Thanks. Looks good. But I can't build it due to -Werror with deprecated functions and incompatible types. Is there a trick to get it built or is it just a intermediate version which doesn't build?

rorybee commented:
@grmpl This is still a work in progress [WIP]. I opened a PR so you can see what I am doing. I will remove the [WIP] in the final working commit.

rorybee committed:
[WIP] Updated app for security

- Fixed deprecation in preferences.xml
- Fixed deprecated PreferenceActivity
- Fixed deprecated Environment.getExternalStorageDirectory
- Fixed duplicate keys in xml

rorybee committed:
Updated app for security

- Added Activity Recognition permission
- Added requesting permissions
- Added permission strings

rorybee commented:
@grmpl It's ready to be merged! :)
